### PR TITLE
WSTEAM1-827: Add EMP core work.

### PR DESCRIPTION
**Pull request by shayneahchoon**
Resolves [WSTEAM1-827](https://jira.dev.bbc.co.uk/browse/WSTEAM1-827)

Overall changes
======
Adds a dynamic Media Player component that loads the correct embedded media player for a user device. 
Article Page:
Screenshot 2023-12-22 at 11 51 17

Live Page: 
Screenshot 2023-12-28 at 16 14 04

Code changes
======
- _An EMP loader has been added here `src/app/components/MediaPlayer/index.tsx`_
- _A BUMP settings builder has been added here `/src/app/components/MediaPlayer/utils/buildSettings.ts`_
- _Parameter extractor functions have been modified to fit BUMP requirements here `src/app/legacy/containers/MediaPlayer/helpers/propsInference/default/index.js`_
- _Added the EMP loader to `LivePage`_

Testing
======
1. _For localhost:_
2. _In your terminal, run `sudo -- sh -c -e "echo '127.0.0.1       localhost.bbc.com' >> /etc/hosts";`._
3. _For the Express app: Go to `http://localhost.bbc.com:7080/afaanoromoo/articles/c4g19kgl85ko` and check that the video loads._
4. _For the NextJS app: Go to `http://localhost.bbc.com:7081/pidgin/live/c7p765ynk9qt?renderer_env=test` and check that the video loads._

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

File: src/app/components/MediaPlayer/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/* eslint-disable import/order */
```
Comment (karinathomasbbc) [inline src/app/components/MediaPlayer/index.test.tsx]: Can we remove this and fix the import order problem?

File: src/app/components/MediaPlayer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
+/* eslint-disable import/order */
+/* eslint-disable @typescript-eslint/ban-ts-comment */
```
Comment (karinathomasbbc) [inline src/app/components/MediaPlayer/index.tsx]: Can we remove these, and fix the underlying issues?

File: src/app/components/MediaPlayer/index.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+};
+
+const Placeholder = ({ setter }: { setter: (value: boolean) => void }) => {
+  return (
+    <button
+      type="button"
+      data-e2e="media-player__placeholder"
+      onClick={() => setter(false)}
+    >
+      TODO: CLICK TO SEE VIDEO
```
Comment (karinathomasbbc) [inline src/app/components/MediaPlayer/index.tsx]: Are we planning to replace this will real / translated placeholder text?

File: src/app/components/MediaPlayer/types.d.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+};
+
+export type BuildConfigProps = {
+  id: string | null;
+  pageType: string;
+  blocks: MediaBlock[];
+  counterName: string | null;
+};
+
+declare global {
```
Comment (karinathomasbbc) [inline src/app/components/MediaPlayer/types.d.ts]: Do we usually put the global types elsewhere? https://github.com/bbc/simorgh/blob/857f0ba17f31d6b34b7821ed3307f256319d1de9/src/global.d.ts

File: src/app/components/MediaPlayer/utils/buildSetting.test.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        holdingImageURL:
+          'https://ichef.test.bbci.co.uk/images/ic/512xn/p01k6mtv.jpg',
+        items: [{ duration: 191, kind: 'programme', versionID: 'p01k6msp' }],
+        guidance: 'Contains strong language and adult humour.',
+      },
+    });
+  });
+
+  it('Should include the mediator parameter if we are on a test url.', () => {
+    process.env.NODE_ENV = 'development';
```
Comment (karinathomasbbc) [inline src/app/components/MediaPlayer/utils/buildSetting.test.ts]: Should we be deleting this in an `afterEach` just in case it impacts other tests? (Given that the default for a local environment is `development`)

eg. `delete process.env.NODE_ENV`

File: src/app/components/MediaPlayer/utils/buildSettings.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+import onClient from '#app/lib/utilities/onClient';
+import { BuildConfigProps } from '../types.d';
+
+const isTestURL = () => {
+  let isTestRender = false;
+
+  if (onClient()) {
+    const search = new URLSearchParams(window.location.search);
+    isTestRender =
+      process.env.NODE_ENV !== 'production' &&
```
Comment (karinathomasbbc) [inline src/app/components/MediaPlayer/utils/buildSettings.ts]: Does this mean we're restricting to only when we're running with `yarn start` only? Might it be easier if we just used presence of renderer_env = test only?

File: src/app/components/MediaPlayer/utils/buildSettings.ts [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+  const isTest = isTestURL();
+
+  const playlistItem = { versionID: clipId, kind, duration: rawDuration };
+
+  return {
+    product: 'news',
+    superResponsive: true,
+    ...(counterName && { counterName }),
+    ...(isTest && { mediator: { host: 'open.test.bbc.co.uk' } }),
```
Comment (karinathomasbbc) [inline src/app/components/MediaPlayer/utils/buildSettings.ts]: Could we call isTestURL() directly here?

```suggestion
    ...(isTestURL() && { mediator: { host: 'open.test.bbc.co.uk' } }),
```

File: src/app/contexts/RequestContext/index.tsx [hunk 8]
```diff
@@ -65,6 +66,7 @@ type RequestProviderProps = {
   mvtExperiments?: MvtExperiment[] | null;
   variant?: Variants | null;
   isUK?: boolean | null;
+  counterName?: string | null;
```
Comment (karinathomasbbc) [inline src/app/contexts/RequestContext/index.tsx:69]: Is this related to the media player? (I thought `counterName` was analytics related?)

File: src/app/legacy/containers/LivePageMediaPlayer/morphMediaPlayer.jsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import React, { useContext } from 'react';
+import { RequestContext } from '#contexts/RequestContext';
+import { GridItemMediumNoMargin } from '#components/Grid';
+import {
+  mediaPlayerPropTypes,
+  emptyBlockArrayDefaultProps,
+} from '#models/propTypes';
+import MediaPlayerContainer from '../MediaPlayer';
+
+const MorphLivePageMediaPlayer = ({ blocks, className }) => {
```
Comment (karinathomasbbc) [inline src/app/legacy/containers/LivePageMediaPlayer/morphMediaPlayer.jsx]: **Nitpick** Could we use Legacy instead of Morph here?

File: src/app/contexts/RequestContext/index.tsx (hunk 8 above)
Comment (shayneahchoon) [reply src/app/contexts/RequestContext/index.tsx:69]: Heya yess, it is part of analytics, but BUMP needs it to report the correct statistics through echo: https://confluence.dev.bbc.co.uk/display/mp/Getting+the+most+from+your+stats#Gettingthemostfromyourstats-ThestatsObject

File: src/app/components/MediaPlayer/index.tsx (hunk 3 above)
Comment (shayneahchoon) [reply src/app/components/MediaPlayer/index.tsx]: yesss, as part of the ongoing work to deprecate Morph.

File: src/app/legacy/containers/LivePageMediaPlayer/index.jsx [hunk 10]
```diff
@@ -1,24 +1,16 @@
-import React, { useContext } from 'react';
-import { RequestContext } from '#contexts/RequestContext';
-import { GridItemMediumNoMargin } from '#components/Grid';
+/** @jsx jsx */
```
Comment (amoore108) [inline src/app/legacy/containers/LivePageMediaPlayer/index.jsx]: Should maybe elevate this file to TypeScript as we'll be keeping it. The `legacyMediaPlayer` not so much in this instance.

File: ws-nextjs-app/pages/[service]/live/[id]/Post/index.tsx [hunk 11]
```diff
@@ -147,13 +149,20 @@ const PostContent = ({ contentBlocks }: { contentBlocks: OptimoBlock[] }) => {
...
+        <LegacyMediaPlayer
+          blocks={props.blocks}
+          className="mediaStyles"
+          css={styles.bodyMedia}
+        />
+      ) : (
+        <LivePageMediaPlayer
+          blocks={props.blocks}
+          className="mediaStyles"
+          css={styles.bodyMedia}
```
Comment (amoore108) [inline ws-nextjs-app/pages/[service]/live/[id]/Post/index.tsx:163]: Kinda feel like these styles should be on the `LivePageMediaPlayer` component directly, rather than be passed/set as props. Just to make it very clear where they are being set from.

File: src/app/components/MediaPlayer/index.tsx [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+    pageType,
+    blocks,
+  });
+
+  const playerConfig = buildConfig({
+    id,
+    pageType,
+    blocks,
+    counterName,
+  });
```
Comment (amoore108) [inline src/app/components/MediaPlayer/index.tsx]: The `getPlayerProps` function looks to be called again within the `buildConfig` function.

Do you think there is a way we could combine these in 1 function that returns both the settings and things like the captionBlock? Or is that too coupled? Just feels a bit wasteful to call `getPlayerProps` twice.

File: scripts/bundleSize/bundleSizeConfig.js [hunk 13]
```diff
@@ -7,5 +7,5 @@
  * We are allowing a variance of -5 on `MIN_SIZE` and +5 on `MAX_SIZE` to avoid the need for frequent changes, as bundle sizes can fluctuate
  */
 
-export const MIN_SIZE = 485 - 5;
-export const MAX_SIZE = 1250 + 5;
+export const MIN_SIZE = 637 - 5;
```
Comment (karinathomasbbc) [inline scripts/bundleSize/bundleSizeConfig.js:10]: Are we worried about this jump? (And it is because of the fact that we have 2 media players now?)

File: src/app/components/MediaPlayer/index.test.tsx [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+import MediaPlayer from '.';
+import sampleBlocks from './fixture';
+import { render } from '../react-testing-library-with-providers';
+
+jest.mock('react', () => ({
+  ...jest.requireActual('react'),
+  useState: jest.fn(),
+}));
+
+describe('MediaPlayer', () => {
```
Comment (karinathomasbbc) [inline src/app/components/MediaPlayer/index.test.tsx]: I think we might need to `jest.clearAllMocks()` before each test?

File: src/app/components/MediaPlayer/index.tsx [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+        if (playerElementRef && playerElementRef.current && playerConfig) {
+          const mediaPlayer = Bump.player(
+            playerElementRef.current,
+            playerConfig,
+          );
+          mediaPlayer.load();
+        }
+      });
+    } catch (error) {
+      logger.error('Failed to bind SMP', error);
```
Comment (karinathomasbbc) [inline src/app/components/MediaPlayer/index.tsx]: Is this the preferred format of an error message - looking at other examples, we use a structured format instead e.g. https://github.com/bbc/simorgh/blob/af5d6b985e4c2c6b44332b13064c72561d303c7a/src/app/legacy/containers/CpsRecommendations/ErrorBoundary.tsx#L31-L33

Actually just realising that this is on the client, so will only appear in the browser console, but perhaps best to be consistent, and follow the established patterns for logging (which may be copy / pasta'ed in future)

File: src/app/components/MediaPlayer/types.ts [hunk 16]
```diff
@@ -0,0 +1,54 @@
+export type PlayerConfig = {
```
Comment (karinathomasbbc) [inline src/app/components/MediaPlayer/types.ts]: Not one for now, I would usually use an `interface` instead of a `type` here. However, I think nowadays, they can be used interchangibly (although type might not allow inheritance whereas interface does). I don't think we have a standard way of doing this.

e.g. [Curation Interface](https://github.com/bbc/simorgh/blob/af5d6b985e4c2c6b44332b13064c72561d303c7a/src/app/models/types/curationData.ts#L52-L65) vs [Environment type](https://github.com/bbc/simorgh/blob/af5d6b985e4c2c6b44332b13064c72561d303c7a/src/app/models/types/global.ts#L3)

But let's leave it, perhaps a discussion for a future Tech/Security session, or worth adding something to the Component Standards? (cc @amoore108 @andrewscfc)

File: src/app/components/MediaPlayer/utils/buildSetting.test.ts [hunk 17]
```diff
@@ -0,0 +1,90 @@
+import { MediaBlock } from '../types';
+import buildConfig from './buildSettings';
+import blocks from '../fixture';
+
+describe('buildSettings', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
```
Comment (karinathomasbbc) [inline src/app/components/MediaPlayer/utils/buildSetting.test.ts]: I seem to recall there being issues with resetAllMocks, and that clearAllMocks works better instead (I don't know why, I just know I have had problems in the past). I think it could be seen if you were to run an individual test vs running them all....

File: src/app/legacy/containers/LivePageMediaPlayer/index.test.tsx [hunk 18]
```diff
@@ -1,17 +1,17 @@
...
-import { LIVE_PAGE } from '#app/routes/utils/pageTypes';
 import { render } from '../../../components/react-testing-library-with-providers';
 import LivePageMediaPlayer from '.';
 import { validLivePageVideoWithCaptionBlock } from '../MediaPlayer/fixtureData';
+import { LIVE_PAGE } from '../../../routes/utils/pageTypes';
 
 const GenerateMediaPlayer = ({
   /* eslint-disable react/prop-types */
   blocks,
   /* eslint-enable react/prop-types */
```
Comment (karinathomasbbc) [inline src/app/legacy/containers/LivePageMediaPlayer/index.test.tsx]: Can we get rid of these linting overrides now that we're in typescript land?

File: src/app/legacy/containers/LivePageMediaPlayer/index.tsx [hunk 19]
```diff
@@ -0,0 +1,20 @@
+/** @jsx jsx */
```
Comment (karinathomasbbc) [inline src/app/legacy/containers/LivePageMediaPlayer/index.tsx]: Do we also need 

`/** @jsxRuntime classic */`?

I seem to remember @amoore108 stating that we needed this for compability between NextJS & Express (but can't recall the detail)?

File: src/app/legacy/containers/LivePageMediaPlayer/index.tsx
Comment (karinathomasbbc) [inline src/app/legacy/containers/LivePageMediaPlayer/index.tsx:1]: Should this be in the src/app/legacy/containers directory now that it has been migrated to TS? Could this be called MediaPlayer and moved to src/app/components?

File: src/app/legacy/containers/LivePageMediaPlayer/legacyMediaPlayer.tsx [hunk 20]
```diff
@@ -0,0 +1,33 @@
+import React, { useContext } from 'react';
+import MediaPlayerContainer from '../MediaPlayer';
+import { RequestContext } from '../../../contexts/RequestContext';
+import { MediaBlock } from '../../../components/MediaPlayer/types';
+import { GridItemMediumNoMargin } from '../../components/Grid';
+
+const LegacyLivePageMediaPlayer = ({ blocks, className }: Props) => {
+  const { id } = useContext(RequestContext);
+
+  if (id === null) return null;
```
Comment (karinathomasbbc) [inline src/app/legacy/containers/LivePageMediaPlayer/legacyMediaPlayer.tsx]: **nitpick** is this easier? 
```suggestion
  if (!id) return null;
```

File: ws-nextjs-app/pages/[service]/live/[id]/Post/index.tsx [hunk 21]
```diff
@@ -9,12 +9,14 @@ import Blocks from '#app/legacy/containers/Blocks';
 import Paragraph from '#app/legacy/containers/Paragraph';
 import UnorderedList from '#app/legacy/containers/BulletedList';
 import LivePageMediaPlayer from '#app/legacy/containers/LivePageMediaPlayer';
+import LegacyMediaPlayer from '#app/legacy/containers/LivePageMediaPlayer/legacyMediaPlayer';
```
Comment (karinathomasbbc) [inline ws-nextjs-app/pages/[service]/live/[id]/Post/index.tsx]: Might we consider renaming the parent component to just `MediaPlayer` instead of `LivePageMediaPlayer`?

File: src/app/components/MediaPlayer/types.ts (hunk 16 above)
Comment (amoore108) [reply src/app/components/MediaPlayer/types.ts]: My personal preference is to use `type` and only use `interface` when needing to use `extend`. `type` can be extended by other type/interface too, but the syntax is less nice imo. But again, just my personal preference.

https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces

File: ws-nextjs-app/pages/[service]/live/[id]/Post/index.tsx (hunk 21 above)
Comment (amoore108) [reply ws-nextjs-app/pages/[service]/live/[id]/Post/index.tsx]: I think the idea we had is that all page types would create their own "wrapper" for the media player if its used in the page, so that custom styling can be more easily applied. This would allow the actual `MediaPlayer` component, which renders the actual player, to be completely generic and agnostic to the where it is placed.

We should probably move the `LivePageMediaPlayer` out of `legacy/containers` though to somewhere more appropriate and closer to the Live Page if this is the pattern we want to follow.

File: scripts/bundleSize/bundleSizeConfig.js (hunk 13 above)
Comment (amoore108) [reply scripts/bundleSize/bundleSizeConfig.js:10]: I don't think this is too much of a concern atm as @eagerterrier is playing around with some bundling strategy changes that will change this figure around a bit.

File: src/app/legacy/containers/LivePageMediaPlayer/index.tsx (hunk 19 above)
Comment (amoore108) [reply src/app/legacy/containers/LivePageMediaPlayer/index.tsx]: I'll need to double check this. I _think_ there was something to do Opera Mini not playing ball, but this doesn't seem to have been much of a problem lately.

The Next.js app at least should throw a pretty loud error if it comes across a component it can't render under its default runtime.

Comment (amoore108) [reply src/app/legacy/containers/LivePageMediaPlayer/index.tsx]: Forgot I possibly addressed this already: https://github.com/bbc/simorgh/pull/10809/files#diff-ce267c5e6295c50027916a3621a5a2e8525964d5526d3c1a560d52c1eca4e347R1

🤦 I'll get a PR up to remove the other references to `jsxruntime` as I don't think they're needed anymore. Thanks for reminding me of this!

File: src/app/components/MediaPlayer/utils/buildSetting.test.ts (hunk 17 above)
Comment (shayneahchoon) [reply src/app/components/MediaPlayer/utils/buildSetting.test.ts]: Coolio, will use restoreMocks instead as it properly restores the original implementation.

File: src/app/components/LivePageMediaPlayer/index.tsx
Comment (karinathomasbbc) [inline src/app/components/LivePageMediaPlayer/index.tsx:1]: Could this component name be more generic? Or can it only be used on LivePages? Is this a toucan media player?

File: src/app/components/MediaLoader/README.md [hunk 22]
```diff
@@ -0,0 +1,22 @@
+# Media Player
```
Comment (karinathomasbbc) [inline src/app/components/MediaLoader/README.md]: Should this be Media Loader?
```suggestion
# Media Loader
```

File: src/app/components/MediaLoader/README.md [hunk 23]
```diff
@@ -0,0 +1,22 @@
...
+
+## Description
+
+This component loads a media player into our webpage. It uses the BBC Universal Media Player module (BUMP) to choose the correct the Embedded Media Player (EMP) for the user's device. Using BUMP removes considerations towards variations in user device and web-browsers as BUMP handles this logic for us and provides the best fitting EMP for our webpage. For example, BUMP can cater for differences between mobiles, laptops and desktops, but also factors such as whether the user is using network data (3G) or a WIFI. 
+
+
+## Local Development
+So that the EMP can load video data, our localhost's domain name should be altered from `localhost:7080` to `localhost.bbc.com:7080` to fully encorporate the bbc domain name (bbc) and top-level domain (.com).
+
+To do, run the command:
```
Comment (karinathomasbbc) [inline src/app/components/MediaLoader/README.md]: Something along these lines, so that folk don't run it multiple times?
```suggestion
First, run the command (you only need to run this once): 
```

File: src/app/components/MediaLoader/utils/buildSettings.ts [hunk 24]
```diff
@@ -0,0 +1,60 @@
+import getPlayerProps from '#app/legacy/containers/MediaPlayer/helpers/propsInference';
+import onClient from '#app/lib/utilities/onClient';
+import { BuildConfigProps } from '../types';
+
+const isTestURL = () => {
+  let isTestRender = false;
+
+  if (onClient()) {
+    const search = new URLSearchParams(window.location.search);
+    isTestRender = search.get('renderer_env') === 'test';
```
Comment (karinathomasbbc) [inline src/app/components/MediaLoader/utils/buildSettings.ts]: **nitpick**: (feel free to ignore)
```suggestion
    const queryParams = new URLSearchParams(window.location.search);
    isTestRender = queryParams.get('renderer_env') === 'test';
```

**Review by karinathomasbbc — APPROVED** [review]

File: ws-nextjs-app/pages/[service]/live/[id]/Post/index.tsx (hunk 11 above)
Comment (amoore108) [reply ws-nextjs-app/pages/[service]/live/[id]/Post/index.tsx:163]: @shayneahchoon Do we need to pass down the `className` and `css` props anymore since `css` is now set directly on the `LivePageMediaPlayer` component>

File: src/app/components/LivePageMediaPlayer/index.tsx
Comment (amoore108) [reply src/app/components/LivePageMediaPlayer/index.tsx:1]: I think if we're keeping the pattern of each page type implementing their own media player, then this should be moved down to the same level as the other Live page specific components, so in the `[service]/live/[id]` directory.

Debatable if this is the approach we want. Will be less DRY, but means we can have complete control over each media player implementation so that we're not doing conditional styling everywhere.

Comment (karinathomasbbc) [reply src/app/components/LivePageMediaPlayer/index.tsx:1]: Happy to leave it as is for now, so that you can progress, but would like to maybe have a wider discussion sometime in the future (if we were to upgrade the media player on one of the other page types perhaps) to decide what the best pattern might be.

Because ideally we could have something like this on any page with a media player (but this might be too simplistic / not achievable):
```
import { GenericMediaPlayer } from '#app/components/MediaPlayer';
import styles from './index.styles';
...

<main>
...
  <MediaPlayer css={styles.myPageMediaPlayer} />
</main>
```

Comment (amoore108) [reply src/app/components/LivePageMediaPlayer/index.tsx:1]: Yea I don't mind that approach, just so long as the styling is de-coupled from the MediaPlayer itself and is implemented by the page or component that is using the MediaPlayer. The wrapping component `LivePageMediaPlayer` may be too much, your approach of just letting the MediaPlayer itself accept CSS is likely the better way to do it.

Comment (amoore108) [reply src/app/components/LivePageMediaPlayer/index.tsx:1]: @shayneahchoon What do you think ☝️ ?

Comment (shayneahchoon) [reply src/app/components/LivePageMediaPlayer/index.tsx:1]: I agree,

Comment (amoore108) [reply src/app/components/LivePageMediaPlayer/index.tsx:1]: Yea this is looking much better https://github.com/bbc/simorgh/pull/11347/commits/8778674139711cbb7c126f7e706f7a1cf186ad3e. Thanks @shayneahchoon , great idea @karinathomasbbc !

**Review by amoore108 — APPROVED** [review]
This looks like a great start, thanks for all the hard work!

File: src/app/components/MediaLoader/utils/buildSettings.ts [hunk 25]
```diff
@@ -0,0 +1,60 @@
...
+const isTestURL = () => {
+  let isTestRender = false;
+
+  if (onClient()) {
+    const queryParams = new URLSearchParams(window.location.search);
+    isTestRender = queryParams.get('renderer_env') === 'test';
+  }
+
+  return isTestRender;
+};
```
Comment (amoore108) [inline src/app/components/MediaLoader/utils/buildSettings.ts]: Just thinking, do we need to use the Test mediator if we're on the Test environment in general? We tend to not use the `renderer_env` flag on Test, as the environment variables should determine what to use behind the scenes.

I'm not 100% sure the videos will work when this is deployed to Test if we aren't using the Test mediator.

Comment (shayneahchoon) [reply src/app/components/MediaLoader/utils/buildSettings.ts]: I put this one in to apply this setting: `{ mediator: { host: 'open.test.bbc.co.uk' } }`

Comment (shayneahchoon) [reply src/app/components/MediaLoader/utils/buildSettings.ts]: I figured that way, we could easily switch between live and test assets when doing development on our localhost.

Comment (shayneahchoon) [reply src/app/components/MediaLoader/utils/buildSettings.ts]: I could switch it the other way round so that `{ mediator: { host: 'open.test.bbc.co.uk' } }` is applied by default, but removed if it's on live?

Comment (amoore108) [reply src/app/components/MediaLoader/utils/buildSettings.ts]: I think its ok for localhost, I'm just thinking when its deployed to Test, will we need to remember to add in `renderer_env=test` to get the video player to work correctly?

Comment (amoore108) [reply src/app/components/MediaLoader/utils/buildSettings.ts]: My gut feeling is that the mediator value should default to the environment its on and then we override it with the `renderer_env` query param.

So on local/test it defaults to `{ mediator: { host: 'open.test.bbc.co.uk' } }` and we use `renderer_env` flags to override it if needed.
On Live `mediator` is removed entirely and cannot be overridden.

Comment (shayneahchoon) [reply src/app/components/MediaLoader/utils/buildSettings.ts]: Coolio

Comment (amoore108) [reply src/app/components/MediaLoader/utils/buildSettings.ts]: Something along these lines may work:
```
switch (true) {
  // Conditions to use Test Mediator
  case queryParams.get('renderer_env') === 'test':
  case isLocal():
  case isTest():
    return true;
  // Conditions to use Live Mediator
  case queryParams.get('renderer_env') === 'live' && !isLive():
    return false;
  default:
    return false;
}
```

The ordering of the cases is important, so the above may not work and require a bit of testing to get it just right. `if statements` maybe clearer for this.

**Review by Isabella-Mitchell — APPROVED** [review]
Looks good - thanks for going through it! A few notes from our call

- Will need to check that this doesn't break if editorial try to upload audio before we support it.
- Will need to update the NextJS threat model with BUMP/ our Runbook before it goes to production.
